hack/windows/Install-OVS.ps1

```diff
@@ -44,48 +31,79 @@ $ErrorActionPreference = "Stop"
+$MininalVCRedistVersion="14.12.25810"
+
 $WorkDir = [System.IO.Path]::GetDirectoryName($myInvocation.MyCommand.Definition)
-$OVSDownloadDir = $WorkDir
+$InstallLog = "$WorkDir\install_ovs.log"
 $PowerShellModuleBase = "C:\Windows\System32\WindowsPowerShell\v1.0\Modules"
+$OVSZip=""
 
-if (!$LocalFile) {
-    $OVSZip = "$OVSDownloadDir\ovs-win64.zip"
-} else {
-    $OVSZip = $LocalFile
-    $DownloadDir = Split-Path -Path $LocalFile
-}
+function PrepareOVSLocalFiles() {
+    if ($LocalFile -ne "") {
+        validateLocalOVSFiles
+        $OVSDownloadURL = ""
+        $OVSPublishedHash = ""
+    } else {
+        $OVSZip = "$WorkDir\ovs-win64.zip"
+        if ($DownloadURL -ne "" -and $DownloadURL -ne "$OVSDownloadURL") {
+            $OVSDownloadURL = $DownloadURL
+            $OVSPublishedHash = ""
+        }
+    }
 
-if ($DownloadDir -ne "") {
-    $OVSDownloadDir = $DownloadDir
+    if (($OVSZip -ne "")) {
+        if (Test-Path -Path $OVSInstallDir) {
+            Log "$OVSInstallDir already exists, exit OVS installation."
+            exit 1
+        }
+        $removeZipFile = $false
+        if ($OVSDownloadURL -ne "") {
+            DownloadOVS
+            $removeZipFile = $true
+        }
+        $ovsInstallParentPath = Split-Path -Path $OVSInstallDir -Parent
+        Expand-Archive -Path $OVSZip -DestinationPath $ovsInstallParentPath | Out-Null
+        if ($removeZipFile) {
+            rm $OVSZip
+        }
+    }
 }
 
-$InstallLog = "$OVSDownloadDir\install_ovs.log"
-
-if ($DownloadURL -ne "") {
-    $OVSDownloadURL = $DownloadURL
-    # For user-provided URLs, do not verify the hash for the archive.
-    $OVSPublishedHash = ""
-}
+function validateLocalOVSFiles() {
+    if (-not (Test-Path $LocalFile)){
+        Log "Path $LocalFile doesn't exist, exit"
+        exit 1
+    }
 
-function Log($Info) {
-    $time = $(get-date -Format g)
-    "$time $Info `n`r" | Tee-Object $InstallLog -Append | Write-Host
-}
+    $ovsFile = Get-Item $LocalFile
+    if ($ovsFile -is [System.IO.FileInfo]) {
+        if ($ovsFile.Attributes -eq "Archive" and ($ovsFile.Extension -eq ".zip" )) {
+            $OVSZip = $LocalFile
+            return
+        }
+        Log "The OVS local package must be in ZIP format, exit"
+        exit 1
+    }
 
-function CreatePath($Path){
-    if ($(Test-Path $Path)) {
-        mv $Path $($Path + "_bak")
+    if ($ovsFile -is [System.IO.DirectoryInfo]) {
+        if ($ovsFile.BaseName -eq "openvswitch") {
+            $OVSInstallDir = $ovsFile.FullName
+            return
+        }
+        Log "Unsupported local directory: $LocalFile. Only directories with a base name 'openvswitch' are supported"
+        exit 1
     }
-    mkdir -p $Path | Out-Null
-}
 
-function SetEnvVar($key, $value) {
-    [Environment]::SetEnvironmentVariable($key, $value, [EnvironmentVariableTarget]::Machine)
+    $attributes = $ovsFile.Attributes
+    Log "Unsupported local file $LocalFile with attributes '$attributes'"
```

Comment:
I don't get here, looks like you didn't check attributes but return unsupported directly.

Reply:
line 82 is checking the attributes, for line 90, the value is Directory. If both the above two cases are not satisfied, the value is not supported.